### DISCUSSION TITLE
ant design是否加入了use client的支持

### DISCUSSION THREAD

**Discussion by skyddyyu** [ant-design/ant-design#45179] [Q&A] [answered]
当我将ant design用于next.js中的服务端组件的时候，比如在服务端组件中使用Table、Tabs等组件可以正常使用。
而我自定义的包含状态useState的组件放入服务端组件就会提示声明为“use client”
之后看ant design的Table、Tabs源码上似乎并没有声明为“use client”
为何ant design的组件不需要“use client”就可以在服务端组件上正常使用😦
这些组件里面应该有状态的

**Comment by kiner-tang** [comment, marked as answer]
是因为 ant design 自己在编译阶段注入了 use client. https://github.com/ant-design/antd-tools/blob/35628fc6c86e62fe6de3bac7d0c01235a1ef05cb/lib/gulpfile.js#L197
编译生成出来的文件实际上都是有加上 'use client' 的。
<img width="973" alt="image" src="https://github.com/ant-design/ant-design/assets/10286961/fadf6ef3-3206-4417-b9a5-aa45d2a4a579">

**Reply by skyddyyu** [reply]
感谢热心解答🫡

**Comment by jxccc1998** [comment]
所以所有组件都是作为客户端组件被渲染的吗

**Reply by li-jia-nan** [reply]
是的
